### PR TITLE
fix: Add Siswati News Classification

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Cover Siswati in Classification

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: docs/mmteb/points/470.jsonl [hunk 1]
```diff
@@ -1,2 +1,2 @@
 {"GitHub": "imenelydiaker", "New dataset": 2}
-{"GitHub": "isaac-chung", "New dataset": 2}
```
Comment (isaac-chung) [inline docs/mmteb/points/470.jsonl:2]: Was a review instead.

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        description="Siswati News Classification Dataset",
+        reference="https://huggingface.co/datasets/dsfsi/za-isizulu-siswati-news",
+        dataset={
+            "path": "isaacchung/siswati-news",
+            "revision": "f5502326c4e48adc99b18b1582f68b8fb5e7ec30",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["zul-Latn"],
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:21]:
```suggestion
        eval_langs=["ssw-Latn"],
```
Should it be ssw for the language code?

**Review by imenelydiaker — COMMENTED** [review]
Is it the same dataset as the isiZulu task one?

**Comment by isaac-chung** [comment]
Same origin/paper but different csv file.

**Review by imenelydiaker — APPROVED** [review]

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        domains=["News"],
+        task_subtypes=["Topic classification"],
+        license="CC-BY-SA-4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""@article{Madodonga_Marivate_Adendorff_2023, title={Izindaba-Tindzaba: Machine learning news categorisation for Long and Short Text for isiZulu and Siswati}, volume={4}, url={https://upjournals.up.ac.za/index.php/dhasa/article/view/4449}, DOI={10.55492/dhasa.v4i01.4449}, author={Madodonga, Andani and Marivate, Vukosi and Adendorff, Matthew}, year={2023}, month={Jan.} }
+        """,
+        n_samples={"train": 80},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:34]: Is it really only 80?

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+N_SAMPLES = 2800
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:6]: delete?

**Review by KennethEnevoldsen — COMMENTED** [review]
A few late comments here (@isaac-chung just want to ensure that the size is not an error)

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py (hunk 3 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:34]: Yes. Is that too small?

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py (hunk 4 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:6]: Can do in my next PR. Good catch!

File: mteb/tasks/Classification/ssw/SiswatiNewsClassification.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/ssw/SiswatiNewsClassification.py:34]: I don't think so. Seems like the scores are reasonably differentiated. I was just worried that it was a typo

**Pull request merged**